src/Compilers/CSharp/Test/Semantic/Semantics/InterceptorsTests.cs

```diff
@@ -7199,4 +7199,211 @@ static void Main()
+            //     [global::System.Runtime.CompilerServices.InterceptsLocationAttribute(1, "q2jDXUSFcU71GJHh7313cHEAAABQcm9ncmFtLmNz")]
+            Diagnostic(ErrorCode.ERR_InterceptorNameNotInvoked, "global::System.Runtime.CompilerServices.InterceptsLocationAttribute").WithArguments("P").WithLocation(6, 6));
+    }
+
+    [Fact]
+    public void ConditionalAccess_03()
+    {
+        // Conditional access on a nullable value type, original and interceptor use non-nullable value type.
+        // Note that we can't intercept a conditional-access with an extension due to https://github.com/dotnet/roslyn/issues/71657
+        var source = CSharpTestSource.Parse("""
+            using System;
+
+            partial struct S
+            {
+                void M() => throw null!;
+
+                static void Main()
+                {
+                    S? s = null;
+                    s?.M();
+                    Console.Write(1);
+                }
+            }
+            """, "Program.cs", RegularWithInterceptors);
+
+        var comp = CreateCompilation(source);
+        var model = comp.GetSemanticModel(source);
+        var node = source.GetRoot().DescendantNodes().OfType<InvocationExpressionSyntax>().First();
+        var locationSpecifier = model.GetInterceptableLocation(node)!;
+
+        var interceptors = CSharpTestSource.Parse($$"""
+            partial struct S
+            {
+                {{locationSpecifier.GetInterceptsLocationAttributeSyntax()}}
+                public void M1() { }
+            }
+            """, "Interceptors.cs", RegularWithInterceptors);
+
+        var verifier = CompileAndVerify([source, interceptors, s_attributesTree], expectedOutput: "1");
+        verifier.VerifyDiagnostics();
+    }
+
+    [Theory]
+    [InlineData("p->M();")]
+    [InlineData("(*p).M();")]
+    public void PointerAccess_01(string invocation)
+    {
+        var source = CSharpTestSource.Parse($$"""
+            struct S
+            {
+                void M() => throw null!;
+
+                static unsafe void Main()
+                {
+                    S s = default;
+                    S* p = &s;
+                    {{invocation}}
+                }
+            }
+            """, "Program.cs", RegularWithInterceptors);
+
+        var comp = CreateCompilation(source, options: TestOptions.UnsafeDebugExe);
+        CompileAndVerify(comp, verify: Verification.Fails);
+
+        var model = comp.GetSemanticModel(source);
+        var node = source.GetRoot().DescendantNodes().OfType<InvocationExpressionSyntax>().Single();
+        var locationSpecifier = model.GetInterceptableLocation(node)!;
+
+        var interceptors = CSharpTestSource.Parse($$"""
+            #nullable enable
+            using System;
+
+            static class Interceptors
+            {
+                {{locationSpecifier.GetInterceptsLocationAttributeSyntax()}}
+                public static void M1(this ref S s) => Console.Write(1);
```

Comment:
> void M1(this ref S s) [](http://example.com/codeflow?start=29&length=22)

Do we need to expand the "signature matching" portion of the spec to accommodate for this?

Reply:
I think these cases are covered (and enforced) by the following from the spec

> - When an interceptable instance method is compared to a static interceptor method (including a classic extension method), we use the method as if it is an extension in reduced form for comparison. The first parameter of the static method is compared to the instance method `this` parameter.
> - The returns and parameters, including the `this` parameter, must have the same ref kinds and types.